### PR TITLE
[Rector] Update rector and re-run with LevelSetList::UP_TO_PHP_74

### PR DESCRIPTION
**Pull request by samsonasik**
| Q             | A
| ------------- | ---
| QA?       | ✔️

This PR do the followings:

- Update rector dependency to latest 0.12.21
- Update LevelSetList to `LevelSetList::UP_TO_PHP_74` as current composer.json require php 7.4, and run it.
- Set auto import option to make typed property use its name instead of fqcn when possible.

Note: For update to typed property, it only update private property only when possible to avoid BC break.

### CODE REVIEW

**Comment by samsonasik** [comment]
It is ready for review 🎉 

- psalm error is unrelated as happen in latest master as well ref https://github.com/spiral/framework/runs/5873145374?check_suite_focus=true#step:10:13
- build on php 8.1 is unrelated as happen in latest master as well ref https://github.com/spiral/framework/runs/5873146180?check_suite_focus=true#step:11:8

**Comment by samsonasik** [comment]
updated rector to 0.12.21 and re-run it

**Comment by samsonasik** [comment]
updated rector to 0.12.22 and re-run it

**Comment by samsonasik** [comment]
rebased.

**Comment by samsonasik** [comment]
updated rector to 0.12.23 and re-run it

**Comment by samsonasik** [comment]
rebased and re-run rector

**Comment by samsonasik** [comment]
rebased and re-run rector

**Comment by samsonasik** [comment]
updated to rector 0.13.0

**Comment by samsonasik** [comment]
I updated to rector 0.13.2 and re-run rector

**Comment by samsonasik** [comment]
I updated to rector 0.13.4 and re-run rector

**Comment by samsonasik** [comment]
updated to rector 0.13.5

**Comment by samsonasik** [comment]
I updated to rector 0.13.8 and re-run rector

File: src/Attributes/src/Internal/Decorator.php [hunk 1]
```diff
@@ -33,39 +35,39 @@ public function __construct(ReaderInterface $reader)
...
+    public function getPropertyMetadata(ReflectionProperty $property, string $name = null): iterable
     {
         return $this->reader->getPropertyMetadata($property, $name);
     }
 
     /**
      * {@inheritDoc}
      */
-    public function getConstantMetadata(\ReflectionClassConstant $constant, string $name = null): iterable
+    public function getConstantMetadata(ReflectionClassConstant $constant, string $name = null): iterable
```
Comment (roxblnfk) [inline src/Attributes/src/Internal/Decorator.php:62]: Hello.
There are a lot of unnecessary changes. Can you turn off the rule to disable import of internal classes from global namespace?

Comment (samsonasik) [reply src/Attributes/src/Internal/Decorator.php:62]: Yes, I can disable for short import classes:

via:

```php
$rectorConfig->disableImportShortClasses();
```

https://github.com/rectorphp/rector-src/blob/328c43713e7b7ec77675637582117619af012985/packages/Config/RectorConfig.php#L88

I may re-create PR for it when I have a chance

Comment (samsonasik) [reply src/Attributes/src/Internal/Decorator.php:62]: I re-create PR for it at https://github.com/spiral/framework/pull/736

**Comment by samsonasik** [comment]
Closing in favor of https://github.com/spiral/framework/pull/736
